invenio_github/api.py

```diff
@@ -319,39 +317,52 @@ def remove_hook(self, repo_id, name):
                 return True
         return False
 
-    def get_repository_releases(self, repo_name="", repo=None):
-        """Retrieve repository releases."""
-        if not (repo or repo_name):
-            raise ValueError("At least one of (repo, repo_name) is required")
-
-        if not repo:
-            repo = Repository.get(name=repo_name)
+    def get_repository_releases(self, repo=None):
+        """Retrieve repository releases. Returns API release objects."""
+        if not (repo):
+            raise ValueError("Repo is required")
 
-        check_repo_access_permissions(repo, self.user_id, repo.github_id, repo_name)
+        check_repo_access_permissions(repo, self.user_id, repo.github_id, repo.name)
 
         # Retrieve releases and sort them by creation date
-        release_objects = sorted(repo.releases.all(), key=lambda r: r.created)
+        release_instances = []
+        for release_object in repo.releases.order_by(Release.created):
+            release_instance = current_github.release_api_class(release_object)
+            release_instances.append(release_instance)
 
-        return release_objects
+        return release_instances
 
     def get_user_repositories(self):
-        """Retrieves user repositories."""
-        token = self.session_token
-        if token:
-            extra_data = self.account.extra_data
-        repos = extra_data.get("repos", [])
+        """Retrieves user repositories, containing db repositories plus remote repositories."""
+        repos = self.user_remote_repositories
         if repos:
             # 'Enhance' our repos dict, from our database model
-            db_repos = Repository.query.filter(
-                Repository.github_id.in_([int(k) for k in repos.keys()]),
-            ).all()
+            db_repos = self.user_repositories
             for repo in db_repos:
                 repos[str(repo.github_id)]["instance"] = repo
                 repos[str(repo.github_id)]["latest"] = GitHubRelease(
                     repo.latest_release()
                 )
         return repos
 
+    @property
+    def user_repositories(self):
+        """Retrieve user repositories from the model."""
+        db_repos = Repository.query.filter(Repository.user_id == self.user_id).all()
+        return db_repos
+
+    @property
+    def user_remote_repositories(self):
+        """Retrieve user repositories from user's remote data."""
+        extra_data = self.account.extra_data
+        repos = extra_data.get("repos", [])
+        return repos
+
+    def disable_repo(self, repo):
+        """Disables all current user repositories."""
+        assert repo.user_id == self.user_id
```

Comment:
Yes, it comes from the model. Before we were checking it on access (e.g. `Repository.get()`) and that was moved to the business logic. 

This "disable_repo" was moved from the disconnect handler, to wrap the repo functionality in an API call. 

I realised we also have the permission check to access the repo and that can be reused here.

Reply:
However, since we are hiding the enabling/disabling actions behind the API class and checking for permissions it can have an impact on support. E.g. we'd have to impersonate a user to enable/disable the repo